apps/src/code-studio/components/header/SignInCalloutWrapper.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import cookies from 'js-cookie';
+import SignInCallout from './SignInCallout';
+
+const HideSignInCallout = 'hide_signin_callout';
+
+// The use of both session storage and cookies is to check for 1 day
+// and 1 session, and display the callout again only once BOTH have passed.
```

Comment:
I'm curious about why we need to put this value in session storage?

Also, is it possible to set the cookie for something like 20 hours?  I'm thinking of the scenario where a student always does the activity at around the same time each day and wondering if the behavior might be weird if they do a level without seeing the popup and then see it on the next level.  (Ideally I think we would want to reset this in the middle of the night each day.)

Finally, did we decide if we wanted to cap the number of times we show the callout (or just continue to show it once a day)?

Reply:
The spec stated that we wanted both, so if a student opens the same window the next day (example: student works on it at 2, and opens it again at 2:30 the next day), they don't see the message again in the same session. In fact, I think the use case for both is exactly what your example is actually covered by.

Even if the cookies flag expires, the student's session flag would still prevent it from being shown again unless they started a new lesson in a new window.

There was nothing stated in the spec about limiting the number of overall times it is shown, so I haven't included that.

Reply:
I see.  I think the common case might be that students close their browser and open a new one the next day (e.g. this is what they teach my kids to do when using school laptops) but we can start with this and refine it if we get feedback.

Reply:
Hmm I definitely see what you mean re: students are closing down entirely and starting fresh the next day. And I guess in that case, I'm not sure what the most desirable behavior is (to your point earlier: the solution is potentially a midnight refresh time, which isn't currently set up with either of these tools). I'll keep as-is for now, and be ready to strip out session storage/refine cookies if needed!